namer/namer.cc

```diff
@@ -2001,14 +2000,20 @@ class TreeSymbolizer {
         // Prevent a crash by validating the arguments before we do any symbol lookups or manipulations
         if (send->hasPosArgs() || send->hasKwArgs() || send->block() != nullptr) {
             if (send->numPosArgs() > 1) {
-                if (auto e = ctx.beginError(send->loc, core::errors::Namer::InvalidTypeDefinition)) {
-                    e.setHeader("Too many args in type definition");
+                auto extraArgsLoc = send->getPosArg(1).loc().join(send->getPosArg(send->numPosArgs() - 1).loc());
+                if (auto e = ctx.beginError(extraArgsLoc, core::errors::Namer::InvalidTypeDefinition)) {
+                    e.setHeader("Too many arguments in `{}` definition for `{}`", send->fun.show(ctx),
+                                typeName->cnst.show(ctx));
+                    auto firstArgLoc = send->getPosArg(0).loc();
+                    auto argsLoc = send->argsLoc();
+                    auto replacementRange = core::Loc(ctx.file, firstArgLoc.endPos(), argsLoc.endPos());
+                    e.replaceWith("Retain only the first argument", replacementRange, "",
+                                  ctx.locAt(firstArgLoc).source(ctx).value());
```

Comment:
```suggestion
                    e.replaceWith("Retain only the first argument", replacementRange, "");
```

Sorry for the run-around, I didn't see how this was implemented at first (I thought that the replacement range was from the beginPos of the first argument).

Since that's not the case, and the replacement range is from the endPos of the first arg to the endPos of the last arg, we don't need to interpolate anything into the replacement text of `""` and we can thus omit this final `ctx.locAt` argument.